src/plugins/social/plugin.py

```diff
@@ -522,19 +522,14 @@ def _fetch_font_from_google_fonts(self, family: str):
             with requests.get(match) as res:
                 res.raise_for_status()
 
-                # Create a temporary file to download the font
-                with NamedTemporaryFile() as temp:
-                    temp.write(res.content)
-                    temp.flush()
-
-                    # Extract font family name and style
-                    font = ImageFont.truetype(temp.name)
-                    name, style = font.getname()
-                    name = " ".join([name.replace(family, ""), style]).strip()
-
-                    # Move fonts to cache directory
-                    target = os.path.join(path, family, f"{name}.ttf")
-                    copy_file(temp.name, target)
+                # Extract font family name and style
+                font = ImageFont.truetype(BytesIO(res.content))
+                name, style = font.getname()
```

Comment:
Please try putting it in a context manager `with BytesIO(res.content) as file`
I'm not sure if this will lead to the same issue as before but would be nice to manage it by ourselves.

Reply:
The comment was done from the GitHub android app, now rechecking on the phone, I see that the `src/` and `material/` sources don't match.
Martin also typically asks for comments explaining why changes were made and a t.ly link to the issue thread, but I admit I'm not sure if this situation warrants such a thing 😅

Reply:
Thanks. Excellent point about the resource management. It should not cause any problems to put it in, that was only one with`NamedTemporaryFile`. 

Not sure why the files in the different directories are different. I had made the mistake of editing the one in `material`, which was open because I was debugging. Should have switched back to the one in `src`. I then edited the one in `src`, assuming the running `npm start` process would sync that to `material`. I bet I will learn something once I figure out what mistake I made there...

Re the documentation of the changes, I think that such a comment is necessary when the rationale for a code section needs to be recorded. I don't think that is needed here because the intention is the same as before and I don't think there is a danger that it will be changed back to `NamedTemporaryFile`. Happy to put that comment in if @squidfunk disagrees. In general, I do like the extent to which the code is documented, so am all in favor of doing this, just doesn't seem to apply in this case.

Thanks, again, @kamilkrzyskow for the very useful comments!

Reply:
> Martin also typically asks for comments explaining why changes were made and a t.ly link to the issue thread, but I admit I'm not sure if this situation warrants such a thing 😅

I really think this is a good idea here, so that the person working on this section of code doesn't fall into the same pit again. It's a code section that is fragile because of OS differences. What I would document:

- __What does the code do here__ – for instance, passing the stream directly into ImageFont was surprising to me – never would've thought of that and that's good to know
- __Why did we implement it that way__ – a short statement that we used a temporary file before, because the name needs to be extracted from the file, and what that caused + explain how the current code fixes that + link to the issue shortened by [t.ly](https://t.ly/).